src/System.Web.OData/OData/Query/Expressions/ExpressionBinderBase.cs

```diff
@@ -103,6 +103,82 @@ internal ExpressionBinderBase(IEdmModel model, ODataQuerySettings querySettings)
+        /// <returns>The LINQ <see cref="Expression"/> created.</returns>
+        public virtual Expression BindBinaryOperatorNode(BinaryOperatorNode binaryOperatorNode)
+        {
+            Expression left = Bind(binaryOperatorNode.Left);
+            Expression right = Bind(binaryOperatorNode.Right);
+
+            PromoteExpressionTypes(binaryOperatorNode, ref left, ref right);
+
+            // handle null propagation only if either of the operands can be null
+            bool isNullPropagationRequired = QuerySettings.HandleNullPropagation == HandleNullPropagationOption.True && (IsNullable(left.Type) || IsNullable(right.Type));
+            if (isNullPropagationRequired)
+            {
+                // |----------------------------------------------------------------|
+                // |SQL 3VL truth table.                                            |
+                // |----------------------------------------------------------------|
+                // |p       |    q      |    p OR q     |    p AND q    |    p = q  |
+                // |----------------------------------------------------------------|
+                // |True    |   True    |   True        |   True        |   True    |
+                // |True    |   False   |   True        |   False       |   False   |
+                // |True    |   NULL    |   True        |   NULL        |   NULL    |
+                // |False   |   True    |   True        |   False       |   False   |
+                // |False   |   False   |   False       |   False       |   True    |
+                // |False   |   NULL    |   NULL        |   False       |   NULL    |
+                // |NULL    |   True    |   True        |   NULL        |   NULL    |
+                // |NULL    |   False   |   NULL        |   False       |   NULL    |
+                // |NULL    |   NULL    |   Null        |   NULL        |   NULL    |
+                // |--------|-----------|---------------|---------------|-----------|
+
+                // before we start with null propagation, convert the operators to nullable if already not.
+                left = ToNullable(left);
+                right = ToNullable(right);
+
+                bool liftToNull = true;
+                if (left == NullConstant || right == NullConstant)
+                {
+                    liftToNull = false;
+                }
+
+                // Expression trees do a very good job of handling the 3VL truth table if we pass liftToNull true.
+                return CreateBinaryExpression(binaryOperatorNode.OperatorKind, left, right, liftToNull: liftToNull);
+            }
+            else
+            {
+                return CreateBinaryExpression(binaryOperatorNode.OperatorKind, left, right, liftToNull: false);
+            }
+        }
+
+        /// <summary>
+        /// In case of open Edm proeprties (generated by in $apply) with different CLR types promote smaller type to higher one.
```

Comment:
>smaller type to higher one. [](start = 101, length = 27)

smaller/higher?

Reply:
Give some examples. Also mention that this is "implicit conversion".

---
In reply to: [178372170](https://github.com/kosinsky/WebApi/pull/25#discussion_r178372170) [](ancestors = 178372170)